crates/ruff_python_formatter/src/statement/stmt_if.rs

```diff
@@ -96,35 +48,56 @@ impl FormatNodeRule<StmtIf> for FormatStmtIf {
     }
 }
 
-fn else_if(or_else: &Suite) -> Option<IfOrElIf> {
-    if let [Stmt::If(if_stmt)] = or_else.as_slice() {
-        Some(IfOrElIf::ElIf(if_stmt))
-    } else {
-        None
-    }
-}
-
-enum IfOrElIf<'a> {
-    If(&'a StmtIf),
-    ElIf(&'a StmtIf),
-}
-
-impl<'a> IfOrElIf<'a> {
-    const fn statement(&self) -> &'a StmtIf {
-        match self {
-            IfOrElIf::If(statement) => statement,
-            IfOrElIf::ElIf(statement) => statement,
-        }
-    }
+/// Note that this implementation misses the leading newlines before the leading comments because
+/// it does not have access to the last node of the previous branch
+#[derive(Default)]
+pub struct FormatElifElseClause;
 
-    const fn keyword(&self) -> &'static str {
-        match self {
-            IfOrElIf::If(_) => "if",
-            IfOrElIf::ElIf(_) => "elif",
-        }
+impl FormatNodeRule<ElifElseClause> for FormatElifElseClause {
+    fn fmt_fields(&self, item: &ElifElseClause, f: &mut PyFormatter) -> FormatResult<()> {
+        format_elif_else_clause(item, f, None)
```

Comment:
Nit: You could also implement `FormatNodeRuleWithOptions` where the option is the last node.

Reply:
I tried that but the node has a lifetime and afaik the options do not allow lifetimes